### DISCUSSION TITLE
A leading dash (-) in the desired pattern causes ripgrep to think there is no pattern.

### DISCUSSION THREAD

**Discussion by Amanita-muscaria** [BurntSushi/ripgrep#1676] [General] [answered]
#### What version of ripgrep are you using?

12.1.1

#### How did you install ripgrep?

via pacman

#### What operating system are you using ripgrep on?

5.8.5-arch1-1

#### Describe your bug.

A leading dash (-) in the desired pattern causes ripgrep to think there is no pattern.

#### What are the steps to reproduce the behavior?

The easiest is to search through the help output of a command.
e.g. `rg -h | rg "-S"`

#### What is the actual behavior?
```
❯ rg -h | rg "-S"
error: The following required arguments were not provided:
    <PATTERN>

USAGE:
    
    rg [OPTIONS] PATTERN [PATH ...]
    rg [OPTIONS] [-e PATTERN ...] [-f PATTERNFILE ...] [PATH ...]
    rg [OPTIONS] --files [PATH ...]
    rg [OPTIONS] --type-list
    command | rg [OPTIONS] PATTERN

For more information try --help

```
#### What is the expected behavior?

Returns the matched line(s)

**Comment by BurntSushi** [comment, marked as answer]
From the man page (and the `--help` output):

![rg-man-pattern](https://user-images.githubusercontent.com/456674/91996886-7c0cf780-ed07-11ea-96ac-a93ff847d8e5.png)

That is, you want `rg -h | rg -e -S`. Alternatively, you can use the old Unix convention of `--`, where everything after it is always interpreted as a positional argument. e.g., `rg -h | rg -- -S`. Finally one other trick is to tweak the regex syntax to fool the arg parser: `rg -h | rg '[-]S'`.

grep has the same problems and the same work-arounds.

See also #1560, which is a duplicate of this issue.
